Sources/Tracking/CustomerIOImplementation+ScreenViews.swift

```diff
@@ -78,6 +85,24 @@ internal extension UIViewController {
         CustomerIO.shared.automaticScreenView(name: name, data: data)
     }
 
+    func shouldTrackAutomaticScreenviewEvent() -> Bool {
+        guard let diGraph = SdkInitializedUtilImpl().postInitializedData?.diGraph else {
+            return false // SDK not initialized yet. Therefore, we ignore event.
+        }
+
+        let filter: (UIViewController) -> Bool = diGraph.sdkConfig.filterAutoScreenViewEvents ?? { viewController in
+            let deviceInfo = diGraph.deviceInfo
+
+            let doesViewControllerBelongToHostApp = viewController.bundleIdOfView == deviceInfo.customerBundleId
+
+            return doesViewControllerBelongToHostApp
```

Comment:
Can't we just do something like this?

```
func isSwiftUIViewController(_ viewController: UIViewController) -> Bool {
    return viewController is UIHostingController<AnyView>
}
```

Reply:
I think if we are only dynamically checking for UIHostingController, Instead of checking for specific SwiftUI views.

This addresses the concern of dealing with many private SwiftUI view classes. You're only concerned about whether a screen is backed by SwiftUI or not, not the specifics of the view itself. This was the essence of the function we intended to write.

Regarding, 
> there are many more classes that SwiftUI can use. The list of statements could get long

No matter how many classes are there, they should be in `UIHostingController ` so they should be ignored, if its a pure Swift UI-based application then there won't be any UIViewController and hence we shouldn't even care as our method only checks for UIViewController classes?

> This solution adds a cat and mouse risk where each iOS update that comes out might add/remove/modify classes in the framework that we would need to update for.

I think it's been 3-4 years since Swift-UI is out and the `UIHostingController` so that does seem a bit stable route, we can still keep the custom config method that you exposed along with it as a fallback.